### ISSUE TITLE
Error while sending large message to Hub Server even when setting MaxIncomingWebSocketMessageSize to null

### ISSUE THREAD

**Issue by dimon82** [SignalR/SignalR#3818]
So as the title says, I'm not able to pass a relatively large message to the server. I'm trying to send images and was able to do that with the ones that are around 3Mb, but then I had to send 5Mb one and getting this error which doesn't tell me much:

`Microsoft.AspNet.SignalR.Client.HttpClientException: StatusCode: 500, ReasonPhrase: 'Internal Server Error', Version: 1.1, Content: System.Net.Http.StreamContent`

There are also some headers in the error message but I don't think they are relevant.

Setting `MaxIncomingWebSocketMessageSize` to `null` doesn't change anything.

I'm using SignalR 2.2.1.

The weird thing is that the max message size I'm able to send is 3,142,388 bytes (not sure where this limit is coming from), once I add another character to the message, I'm getting the error above.

My code:
```
var hubConnection = new HubConnection("http://localhost:22222/signalr/hubs");
var hubProxy = hubConnection.CreateHubProxy("ServerHub");
await hubConnection.Start();
await hubProxy.Invoke("SendLargeMessage", message);
```

**Comment by moozzyk** [comment]
What transport do you use? - because MaxIncomingWebSocketMessageSize applies only to webSockets. Try debugging the server, 500 is probably a result of an exception so check what exception is being thrown.

**Comment by dimon82** [comment]
@moozzyk so it looks like it's using ServerSentEvents transport so I guess `MaxIncomingWebSocketMessageSize` is useless for me after all. I've enabled SignalR tracing on the server, no errors are being logged.

UPDATE: nothing to do with SignalR, but rather IIS. The default request size is 4MB, so I had to increase it in web.config and it worked. I'm closing this issue. Thanks for your help, it got me thinking in the right direction.
